Basket-Donation-Payments/Basket-Donation-Payments/View Controllers/Payment Flow/PaymentPriceViewController.m

```diff
@@ -23,7 +23,16 @@ - (void)viewDidLoad {
     [super viewDidLoad];
     self.recipientLabel.text = [NSString stringWithFormat:@"to %@", self.basket.name];
     //TODO: add ability to change currency
+    self.priceInputTextField.delegate = self;
+    UITapGestureRecognizer *tap = [[UITapGestureRecognizer alloc] initWithTarget:self.view action:@selector(endEditing:)];
+    [self.view addGestureRecognizer:tap];
```

Comment:
Another idea to reuse the code, also no need to implement this right now, only if you have any time left at the end of next week. You can encapsulate this behavior by defining a `UIViewController` subclass like `FormViewController` and make all these form-like view controllers with multiple text fields extend `FormViewController`. In the `FormViewController` class you can have something like a `- (void)setupWithTextFields:(NSArray<UITextField *> *)textFields` method where you can make the view controller the delegate for all the text fields and add the tap gesture recognizer. You can also implement `textFieldShouldReturn:` in `FormViewController` and do the implement the text field chaining idea I mentioned before using the list of text fields.